pkg/util/util.go

```diff
@@ -113,6 +113,38 @@ func UnArchiveTar(reader io.Reader, destDir string, arg ...string) error {
 	return nil
 }
 
+// ExrtactSingleDirTar unarchives a tar file and streams its files
+// using the specified io.Reader to the specified destination.
+func ExrtactSingleDirTar(reader io.Reader, destDir string, extractDir string) error {
+	glog.V(1).Infof("begin untar directory ...\n")
+	args := "-zxvC"
+	untar := exec.Command("/usr/bin/tar", args, destDir, extractDir)
+	untar.Stdin = reader
+	var errBuf bytes.Buffer
+	untar.Stderr = &errBuf
+	err := untar.Start()
+	if err != nil {
+		return err
+	}
+	err = untar.Wait()
+	if err != nil {
+		glog.V(3).Infof("%s\n", string(errBuf.Bytes()))
+		return err
+	}
+
+	return nil
+}
+
+// UnArchiveSingleDirTar extracts a  extractDir from local tar file to the specified destination.
+func UnArchiveSingleDirTar(filePath, destDir string, extractDir string) error {
```

Comment:
maybe call it 'UnArchiveSingleDirFromLocalTar' to make it more explicit and correlated with UnArchiveLocalTar